src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp

```diff
@@ -250,7 +250,7 @@ void ShenandoahHeapRegion::make_unpinned() {
 
 void ShenandoahHeapRegion::make_cset() {
   shenandoah_assert_heaplocked();
-  reset_age();
+  // Leave age untouched.  We need to consult the age when we are deciding whether to promote evacuated objects.
```

Comment:
Many other state transitions also `reset_age` and I'm not sure it's necessary. Seems sufficient to do it only for `make_trash`.

Reply:
This is the only case that was causing me "trouble".  Here are the other scenarios found by grep:
  shenandoahHeap.cpp: 2559: ShenandoahFinalUpdateRefsUpdateRegionStateClosure::heap_region_do()